### PR TITLE
#602 Registration component and small fixes to navigation hover colors

### PR DESCRIPTION
**Pull request by Aleks1908**
## Remade the Registration component 
Added components for each type of field and reworked the logic of the Registration form. And some small tweaks on the Navigation. Completed with the help of @Pastedko 

Resolves: #602 

## How to verify:

- [ ] Open the RegistrationForm.jsx and remove the check for the feature switches

- [ ] Run the Front-end and the py-api and register an account. 

- [ ] You should receive an email with verification (that will not work as of the moment because the verification page is not merged)

### CODE REVIEW

File: packages/web/src/components/spa/HackAUBG/RegistrationForm/InputComponent.jsx
Comment (r-mishev) [inline packages/web/src/components/spa/HackAUBG/RegistrationForm/InputComponent.jsx:1]: This component can be refactored to not use this many nested if statements.

File: packages/web/src/components/spa/HackAUBG/RegistrationForm/RegistrationForm.jsx [hunk 1]
```diff
@@ -40,640 +41,298 @@ const RegistrationForm = () => {
...
-            if (field[1] === 'True' || field[1] === 'False') {
-                data[field[0]] = field[1] === 'True' ? true : false;
+            for (const [key, value] of Object.entries(values)) {
+                if (value === 'true') {
+                    updatedValues[key] = true;
+                } else if (value === 'false') {
+                    updatedValues[key] = false;
+                } else {
+                    updatedValues[key] = value;
+                }
```
Comment (r-mishev) [inline packages/web/src/components/spa/HackAUBG/RegistrationForm/RegistrationForm.jsx:65]: Why not just

```suggestion
updatedValues[key] = value
```

File: packages/web/src/components/spa/HackAUBG/RegistrationForm/RegistrationForm.jsx [hunk 2]
```diff
@@ -40,640 +41,298 @@ const RegistrationForm = () => {
...
-        setSubmitButtonValue('Submit');
     };
 
-    useEffect(checkButtonAvailability, [Object.keys(errors)]);
-
-    const showButton = () => {
-        if (loadingAnimation) {
-            return <input type="submit" value="LOADING" className="hackaubg-register-btn" />;
+    const display = () => {
+        if (getValues().team == 'true') {
```
Comment (r-mishev) [inline packages/web/src/components/spa/HackAUBG/RegistrationForm/RegistrationForm.jsx:88]: Are you certain it shouldn't be `` === `` ?

**Review by r-mishev — APPROVED** [review]

File: packages/web/src/components/spa/HackAUBG/VerifyAccountPop/VerifyAccount.jsx [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+    const [jwtToken, setJwtToken] = useState('');
     const location = useLocation();
-    const params = new URLSearchParams(location.search);
-    const jwtToken = params.get('jwt_token');
+    const [buttonState, setButtonState] = useState(true);
+
+    useEffect(() => {
+        const params = new URLSearchParams(location.search);
+        if (params.size === 0) {
+            window.location.href = '/hackaubg';
```
Comment (asynchroza) [inline packages/web/src/components/spa/HackAUBG/VerifyAccountPop/VerifyAccount.jsx:16]: Isn't the url already that? No need to change the value.

File: packages/web/src/components/spa/HackAUBG/VerifyAccountPop/VerifyAccount.jsx [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
+        } else {
+            setJwtToken(params.get('jwt_token'));
+        }
+    }, [location]);
```
Comment (asynchroza) [inline packages/web/src/components/spa/HackAUBG/VerifyAccountPop/VerifyAccount.jsx:20]: I would usually suggest extracting those as hooks but at this point just make it work.

File: packages/web/src/components/spa/HackAUBG/VerifyAccountPop/VerifyAccount.jsx (hunk 3 above)
Comment (Aleks1908) [reply packages/web/src/components/spa/HackAUBG/VerifyAccountPop/VerifyAccount.jsx:16]: This is a check if after verify there is a jwt_token. And if there is not and the url looks like this /hackaubg/verify it goes back to /hackaubg

Comment (asynchroza) [reply packages/web/src/components/spa/HackAUBG/VerifyAccountPop/VerifyAccount.jsx:16]: okay, makes sense

**Pull request merged**